Jetcaster/glancewidget/src/main/java/com/example/jetcaster/glancewidget/JetcasterAppWidget.kt

```diff
@@ -0,0 +1,257 @@
+import androidx.glance.LocalSize
+import androidx.glance.appwidget.GlanceAppWidget
+import androidx.glance.appwidget.GlanceAppWidgetReceiver
+import androidx.glance.appwidget.SizeMode
+import androidx.glance.appwidget.components.Scaffold
+import androidx.glance.appwidget.components.SquareIconButton
+import androidx.glance.appwidget.cornerRadius
+import androidx.glance.appwidget.provideContent
+import androidx.glance.background
+import androidx.glance.layout.Alignment
+import androidx.glance.layout.Box
+import androidx.glance.layout.Column
+import androidx.glance.layout.ContentScale
+import androidx.glance.layout.Row
+import androidx.glance.layout.Spacer
+import androidx.glance.layout.fillMaxSize
+import androidx.glance.layout.padding
+import androidx.glance.layout.size
+import androidx.glance.text.FontWeight
+import androidx.glance.text.Text
+import androidx.glance.text.TextStyle
+import androidx.glance.unit.ColorProvider
+import coil.ImageLoader
+import coil.request.ErrorResult
+import coil.request.ImageRequest
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+internal val TAG = "JetcasterAppWidegt"
+
+/**
+ * Implementation of App Widget functionality.
+ */
+class JetcasterAppWidgetReceiver : GlanceAppWidgetReceiver() {
+    override val glanceAppWidget: GlanceAppWidget
+        get() = JetcasterAppWidget()
+
+
+}
+
+data class JetcasterAppWidgetViewState(
+    val trackTitle: String,
+    val trackSubtitle: String,
+    val isPlaying: Boolean,
+    val albumArtUri: String
+)
+
+private object Sizes {
+    val minWidth = 140.dp
+    val smallBucketCutoffWidth = 250.dp // anything from minWidth to this will have no title
+
+    val imageNormal = 80.dp
+    val imageCondensed = 60.dp
+}
+
+private enum class SizeBucket { Invalid, Narrow, Normal }
+
+@Composable
+private fun calculateSizeBucket(): SizeBucket {
+    val size: DpSize = LocalSize.current
+    val width = size.width
+
+    return when {
+        width < Sizes.minWidth -> SizeBucket.Invalid
+        width <= Sizes.smallBucketCutoffWidth -> SizeBucket.Narrow
+        else -> SizeBucket.Normal
+    }
+}
+
+class JetcasterAppWidget() : GlanceAppWidget() {
+    override val sizeMode: SizeMode
+        get() = SizeMode.Exact
+
+    override suspend fun provideGlance(context: Context, id: GlanceId) {
+
+        val testState = JetcasterAppWidgetViewState(
+            trackTitle = "Serifs afoot",
+            trackSubtitle =  "Roboto Beach",
+            isPlaying = false,
+            albumArtUri = "https://upload.wikimedia.org/wikipedia/commons/thumb/d/d2/Longsbear.JPG/320px-Longsbear.JPG"
+        )
+
+
+        provideContent {
+            val sizeBucket = calculateSizeBucket()
+            val playPauseIcon = if (testState.isPlaying) PlayPauseIcon.Pause else PlayPauseIcon.Play
+            val artUri = Uri.parse(testState.albumArtUri)
+
+            when (sizeBucket) {
+                SizeBucket.Invalid -> WidgetUiInvalidSize()
```

Comment:
Provide default Modifiers for every composable here as per goo.gle/compose-api-guidelines

Reply:
These are top level composables, so it only makes sense to have them filling the maximum size